preprocess/project.go

```diff
@@ -1,29 +1,66 @@
 package preprocess
 
 import (
+	"encoding/json"
 	"fmt"
+	// "io/ioutil"
 	"strconv"
 
 	"github.com/docker/libcompose/config"
+	"github.com/rancher/rancher-compose/utils"
 )
 
-func PreprocessServiceMap(serviceMap config.RawServiceMap) (config.RawServiceMap, error) {
-	newServiceMap := make(config.RawServiceMap)
+type BindingProperty struct {
+	Services map[string]Service `json"services"`
+}
 
-	for k, v := range serviceMap {
-		newServiceMap[k] = make(config.RawService)
+type Service struct {
+	Labels map[string]interface{} `json"labels"`
+	Ports  []interface{}          `json"ports"`
+	Scale  interface{}            `json"scale"`
+}
+
+func PreprocessServiceMap(bindingsBytes []byte) func(serviceMap config.RawServiceMap) (config.RawServiceMap, error) {
+	return func(serviceMap config.RawServiceMap) (config.RawServiceMap, error) {
+		newServiceMap := make(config.RawServiceMap)
+
+		var binding BindingProperty
+		var bindingsServices []string
+		err := json.Unmarshal(bindingsBytes, &binding)
+		if err != nil {
+			return nil, err
+		}
+
+		for k := range binding.Services {
+			bindingsServices = append(bindingsServices, k)
+		}
+
+		for k, v := range serviceMap {
+			newServiceMap[k] = make(config.RawService)
+			if utils.Contains(bindingsServices, k) == true {
+				labelMap := make(map[interface{}]interface{})
+				for key, value := range binding.Services[k].Labels {
+					labelMap[interface{}(key)] = value
+				}
+				v["labels"] = labelMap
+				if ports, portsPresent := v["ports"]; portsPresent {
```

Comment:
I think the idea is to set the ports regardless of whether or not they already existed in the config, so this check can probably be skipped.